### PR TITLE
 Adds linkerd support, updates cpu/memory query

### PR DESCRIPTION
**Pull request by askmeegs**
1. `perf/benchmark/linkerd` provides instructions on how to do the Istio performance with linkerd instead.  (**Note** -- currently, using linkerd with the scripts only works for latency, and for `both` mode.) 

2. Updates `perf/benchmark/runner/prom.py` queries for CPU/Memory to avoid double-counting. (`job=kubernetes-cadvisor`).

### CODE REVIEW

File: perf/benchmark/linkerd/fortio.yaml [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  sessionAffinity: None
+  type: LoadBalancer
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: fortioserver
+  namespace: twopods
+spec:
+  clusterIP: 10.4.29.218
```
Comment (mandarjog) [inline perf/benchmark/linkerd/fortio.yaml]: Why specific ip?

File: perf/benchmark/linkerd/fortio.yaml [hunk 2]
```diff
@@ -0,0 +1,200 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
```
Comment (mandarjog) [inline perf/benchmark/linkerd/fortio.yaml]: This file seems identical to the one in benchmarks.
Can we reuse it? If it is different I would like to know how?

File: perf/benchmark/runner/graph.py [hunk 3]
```diff
@@ -11,97 +11,121 @@
...
+
 
     # 4. prep file-write 
     fn = "".join(title.split())
     f = "/tmp/" + fn + ".html" 
     output_file(f) 
 
     # 5. create chart -> save as interactive HTML 
-    p = build_chart(title, metric, c, series)  
+    p = build_chart(title, x_label, x_series, y_label, y_series)
```
Comment (mandarjog) [inline perf/benchmark/runner/graph.py]: While you are here can you fix the file extensions of the output?
They should be .csv and .json I don’t think that is happening.

File: perf/benchmark/runner/prom.py [hunk 4]
```diff
@@ -60,13 +60,13 @@ def fetch(self, query, groupby=None, xform=None):
...
         return self.fetch(
-            'irate(container_cpu_usage_seconds_total{container_name=~"mixer|policy|discovery|istio-proxy|captured|uncaptured"}[1m])',
+            'irate(container_cpu_usage_seconds_total{job="kubernetes-cadvisor",container_name=~"mixer|policy|discovery|istio-proxy|captured|uncaptured"}[1m])',            
             metric_by_deployment_by_container,
             to_miliCpus)
 
     def fetch_memory_by_container(self):
         return self.fetch(
-            'container_memory_usage_bytes{container_name=~"mixer|policy|discovery|istio-proxy|captured|uncaptured"}',
+           'container_memory_usage_bytes{container_name=~"mixer|policy|discovery|istio-proxy|captured|uncaptured"}',
```
Comment (mandarjog) [inline perf/benchmark/runner/prom.py]: Job = cadvisor?

File: perf/benchmark/README.md [hunk 5]
```diff
@@ -71,18 +73,20 @@ The test has three sidecar modes:
...
 
 Optional flags:
 
 ```
+  -h, --help          show this help message and exit
   --size SIZE         size of the payload
   --client CLIENT     where to run the test from
   --server SERVER     pod ip of the server
   --perf              also run perf and produce flamegraph
+  --linkerd LINKERD   linkerd mode
```
Comment (howardjohn) [inline perf/benchmark/README.md]: Should it just be ` --linkerd    ...   linkerd mode` or does it take an argument?

File: perf/benchmark/linkerd/fortio.yaml (hunk 2 above)
Comment (askmeegs) [reply perf/benchmark/linkerd/fortio.yaml]: The deployment has linkerd injection enabled. https://github.com/askmeegs/tools/blob/linkerd-install/perf/benchmark/linkerd/fortio.yaml#L15

File: perf/benchmark/README.md (hunk 5 above)
Comment (askmeegs) [reply perf/benchmark/README.md]: it doesn't take an argument, it's a bool. https://github.com/askmeegs/tools/blob/linkerd-install/perf/benchmark/runner/runner.py#L238 

should I change the format of the flag?

File: perf/benchmark/linkerd/fortio.yaml (hunk 2 above)
Comment (mandarjog) [reply perf/benchmark/linkerd/fortio.yaml]: https://github.com/istio/tools/blob/master/perf/benchmark/templates/fortio.yaml#L87   It is already there as an option in the helm template.

**Pull request merged**